intervaltree/intervaltree.py

```diff
@@ -437,6 +437,17 @@ def intersection_update(self, other):
             if iv not in other:
                 self.remove(iv)
 
+    def overlap_intervals(self,other):
+        """
+        Returns a new IntervalTree consisting of intervals representing the
+        regions overlapped by at least one interval in both of self and other.
+        """
+        splits = (self | other)
+        splits.split_overlaps()
```

Comment:
Splitting the overlaps is currently very inefficient, O(n^2logn) it should be possible to calculate interval intersections without this step.

Reply:
@chaimleib Hello, I am really interested in this method.
Do you have any hints for finding interval intersections without splitting the overlaps?

Reply:
Iterating through the boundary tables and testing with `a.overlaps(point) and b.overlaps(point)` should yield worst case O( (n+m)*log(n+m) ).